### ISSUE TITLE
settings UI fails to open if browser.tabs.loadDivertedInBackground = true

### ISSUE THREAD

**Issue by atomGit** [wooque/site-bleacher#3]
if browser.tabs.loadDivertedInBackground = true then the settings UI won't open when clicking the "settings" context menu item in the tool-bar icon

FF 65.0.1 on Linux
SB 1.0.18

**Comment by wooque** [comment]
Could not reproduce, same version of Firefox on Linux and same version of Site Bleacher, tried on clean profile also, works fine. We are talking about this item, do we?
![2019-02-25-185351_1920x1080_scrot](https://user-images.githubusercontent.com/1936195/53357785-f1d41e00-392e-11e9-8ee7-85e746426f29.png)
Did you do some other about:config changes or some specific preference changes?

**Comment by atomGit** [comment]
yes, the settings item in the tool bar pop-up menu

and no, no other config changes - i tested on a fresh profile and it worked fine until i set browser.tabs.loadDivertedInBackground = true

**Comment by wooque** [comment]
Could you please check the steps I've done, and if they lead to Settings not opening:

Run `firefox --new-instance --profile $(mktemp -d)`
Go to https://addons.mozilla.org/en-US/firefox/addon/site-bleacher/ and install add-on
Click `Settings` in add-on popup and check if it opens settings page
Go to `about:config` and change `browser.tabs.loadDivertedInBackground` to `true`
Click `Settings` again to check if it opens settings page

It worked fine, maybe it doesn't open if on certain page or something?

**Comment by atomGit** [comment]
i preformed the steps precisely as you instructed and the result was the same - once i set 'browser.tabs.loadDivertedInBackground' to 'true', the settings UI could not be opened from the tool-bar pop-up menu

the way i tested before was essentially the same, except i created my test profile from about:profiles instead of the terminal

to me this is not an important issue, so unless others complain, maybe it isn't worth spending a lot of time troubleshooting

the only thing i can think of that may be different in our environments are a few Firefox settings that may be set by the OS, in my case Manjaro - i know at least some Linux distro's (Mint and Manjaro for sure) make a few adjustments to a config file somewhere that affects all instances of Firefox - i forget if it's a .ini or config file in /var/ somewhere or what

**Comment by atomGit** [comment]
whoops - i should have included that, yes, the settings UI did open after i first installed the add-on (but failed to open only after i set browser.tabs.loadDivertedInBackground to true, as i said)

**Comment by wooque** [comment]
I doubt that distro specific settings are the issue, but I would like to test it.
What version/edition of Manjaro are you running?

**Comment by atomGit** [comment]
Manjaro 18.0.3 / KDE

the files i'm thing of are in /usr, not /var :)

this is a portion of /usr/lib/firefox/browser/defaults/preferences/vendor.js that may be interesting...

```
// Don't disable our bundled extensions in the application directory
pref("extensions.autoDisableScopes", 11);
pref("extensions.shownSelectionUI", true);
```

also in /usr/lib/firefox/browser/features - i `sudo rm *.xpi` all of the system add-ons (privacy reasons/data leakage) - this is the only change i make to the core installation 

there's several other platform specific config files as well, but i don't think they would cause this issue

**Comment by atomGit** [comment]
tested those 2 prefs myself - has no effect

**Comment by atomGit** [comment]
ok, i made a huge error ... if i load up a new profile and set 'browser.tabs.loadDivertedInBackground' to 'true' the settings UI **DOES** appear, but it opens in a new, background tab and i just didn't notice it

still, it does not appear with my default profile, so there's another setting that's involved which i will hunt down shortly

looks like i have caused you a headache for nothing at this point

**Comment by atomGit** [comment]
got it!

browser.tabs.loadDivertedInBackground = true
browser.link.open_newwindow = 1

if both of those are set, then the settings page doesn't open

probably pretty rare that this would be the case since neither of these values are the default

**Comment by wooque** [comment]
Thanks, I managed to reproduce the issue, fix is in 1.0.20 (441df9e29a0336917c46ad5e552d6624c50fd660)
